applications/mlflow/Dockerfile

```diff
@@ -0,0 +1,93 @@
+FROM ubuntu:22.04
+
+# Set environment variables
+ENV DEBIAN_FRONTEND=noninteractive \
+    SHELL=/bin/bash \
+    HOME=/home/devuser
+
+# Install common utilities and prerequisites
+RUN apt-get update && apt-get install -y \
+    curl \
+    git \
+    gnupg \
+    jq \
+    lsb-release \
+    lsof \
+    make \
+    netcat \
+    python3 \
+    python3-pip \
+    python3-venv \
+    software-properties-common \
+    sudo \
+    tar \
+    unzip \
+    vim \
+    wget \
+    && rm -rf /var/lib/apt/lists/* \
+    # Create a symlink from python3 to python for compatibility
+    && ln -sf /usr/bin/python3 /usr/bin/python
+
+# Install CLI tools
+RUN curl https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3 | bash \
+    # Install kubectl
+    && curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl" \
+    && chmod +x kubectl \
+    && mv kubectl /usr/local/bin/ \
+    # Install Kind
+    && curl -Lo /usr/local/bin/kind https://kind.sigs.k8s.io/dl/v0.20.0/kind-linux-amd64 \
+    && chmod +x /usr/local/bin/kind \
+    # Install yq
+    && wget https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64 -O /usr/local/bin/yq \
+    && chmod +x /usr/local/bin/yq \
+    # Install Task
+    && sh -c "$(curl --location https://taskfile.dev/install.sh)" -- -d -b /usr/local/bin \
+    # Install Replicated CLI
+    && curl -s https://api.github.com/repos/replicatedhq/replicated/releases/latest \
+       | grep "browser_download_url.*linux_amd64.tar.gz" \
+       | cut -d : -f 2,3 \
+       | tr -d \" \
+       | wget -qi - \
+    && tar xf replicated_*_linux_amd64.tar.gz replicated \
+    && mv replicated /usr/local/bin/ \
+    && rm replicated_*_linux_amd64.tar.gz
+
+# Install Python dependencies directly
+RUN pip3 install --upgrade pip wheel setuptools \
```

Comment:
`uv pip install` is the new standard I'd use uv for everything these days with python.